### DISCUSSION TITLE
How can I get content of epub book and show it in SwiftUI app?

### DISCUSSION THREAD

**Discussion by moonvader** [witekbobrowski/EPUBKit#13] [General]
I want to show book's content in iOS app (SwiftUI) and also add speech-to-text feature, how it can be done?
Can't find anything about showing or even accessing content in documentation.
Thank you

**Comment by NullTar** [comment]
Hello moonvader
I checked the library but didn't find any code that can parse the content. It seems that only the structure can be parsed. If you see this, please tell me where the part that parses the content is. At the same time, it is not easy to use this library if you want to package the edited text into an epub.

**Comment by witekbobrowski** [comment]
Oh, I have forgot to reply. whooops.

This library does not provide you with a UI to render the epub contents, rather it helps you to access all of the EPUB components and files directly from your swift code. you can think of it as more of a model layer.

**Reply by NullTar** [reply]
Hello witekbobrowski
I think reading text and translating content to epub are quite important.
I'm a beginner in swift and can't be of much help. I hope these two functions can be supported in the future ❤️

**Reply by witekbobrowski** [reply]
I do not believe this library will ever support this, as it accounts for approximately 90% of the work compared to the current library’s content. However, I believe there are alternative libraries that could assist you in achieving your objectives.
